### ISSUE TITLE
0.55.0 crash

### ISSUE THREAD

**Issue by l7s** [crate/crate#3698]
Hi,

We just upgraded one of our clusters to 0.55.0, but unfortunately nodes are crashing after around 45 minutes with dumps as in [hs_err_pid27312.log](https://dl.dropboxusercontent.com/u/3182365/crate/hs_err_pid27312.log) and [hs_err_pid15161.log](https://dl.dropboxusercontent.com/u/3182365/crate/hs_err_pid15161.log)

Regards,
Chris

**Comment by joemoe** [comment]
hi @l7s, thanks for reporting. we'll have a look at it right now. cheers, johannes

**Comment by mikethebeer** [comment]
Hi @l7s,

the posted JVM log dumps give not enough information for us to figure out what exactly causes the crash. Can you provide us the log-file that Crate produces in your configured log path? Per default it is located in `/var/log/crate.log`. Then we might se the root cause.

Thanks,
Michael

**Comment by l7s** [comment]
Hi Michael,

If you email chris [at] level7systems [dot] co [dot] uk with your IP address I will add it to the firewall and give you SSH access to this machines, so you will be able to review the logs and other files.

Regards,
Chris

**Comment by mikethebeer** [comment]
Hi @l7s Chris,

I think I have figured out what caused the issue.

1. Ensure that you set these environment variables in your init.d script:
```sh
MAX_OPEN_FILES=65535
MAX_LOCKED_MEMORY=unlimited
```
This caused this Warning:
```
[2016-06-26 17:04:18,003][WARN ][env                      ] [crate-fr2] max file descriptors [4096] for elasticsearch process likely too low, consider increasing to at least [65536]
```
You still need to add this in your init-script.

2. This is the main reason, why your cluster service always adds and removes nodes. This might caused of a slow or congested network. Per default this setting is set to `3s`. I have set this now on every node to `30s`.

`discovery.zen.ping_timeout: 30s`

The cluster seems stable now.
